src/test/java/com/puppycrawl/tools/checkstyle/ConfigurationLoaderTest.java

```diff
@@ -769,4 +769,20 @@ public void testLoadConfiguration3() throws Exception {
                     .contains("query");
         }
     }
+
+    @Test
+    public void testLengthInReplaceProperties() throws Exception {
+        final String[][] testValues = {
+                {"${a}"},
+                {"x${a}"},
+        };
+        final Properties props = new Properties();
+        for (String[] testValue : testValues) {
+            final String value = (String) getReplacePropertiesMethod().invoke(
+                    null, testValue[0], new PropertiesExpander(props), "Default");
+            assertWithMessage("\"" + testValue[0] + "\"")
+                    .that(value)
+                    .isEqualTo("Default");
```

Comment:
Why `{"x${a}"}` becomes Default? 
Should  be `Default`.

Reply:
The default value is provided as "Default" thats why

Reply:
Should  be `xDefault`.
As x is before variable.

Reply:
I recalling this as default is for whole property value